contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -2103,6 +2121,32 @@ tsql_select_common_typmod_hook(ParseState *pstate, List *exprs, Oid common_type)
 	{
 		Node *expr = (Node*) lfirst(lc);
 		int32 typmod = exprTypmod(expr);
+		Oid   type = exprType(expr);
+		Oid   tsql_type = get_immediate_base_type_of_UDT_internal(type);
+
+		/* 
+		 * Handling for UDT, If the base_type is different
```

Comment:
You mean when `tsql_type` is valid oid that means it is UDT and we need to find its typmod for hanlding? Lets update the comment, its quite confusing.